pkg/controller/vitessshard/reconcile_rollout.go

```diff
@@ -0,0 +1,214 @@
+package vitessshard
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apilabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/pkg/operator/results"
+	"planetscale.dev/vitess-operator/pkg/operator/rollout"
+)
+
+func (r *ReconcileVitessShard) reconcileRollout(ctx context.Context, vts *planetscalev2.VitessShard) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
+
+	if !rollout.Scheduled(vts) {
```

Comment:
This is checking whether the VitessShard's parent wants to apply changes to the VitessShard, which is not what we're interested in here. We want to handle the case after changes have been applied to the VitessShard, and the shard controller needs to then roll out changes to its child tablets.

There is not yet an annotation defined for triggering rolling restart. We should add one in the `rollout` package along with the other annotations. Maybe we can call the new annotation `rollout.planetscale.com/propagate` since it directs the object to propagate the rollout to its children?

We can then change this to check for `if !rollout.Propagated(vts) {`. Once we find that there are no children left with scheduled changes, we can remove the annotation with `rollout.Unpropagate(vts)` since we are done at that point.

Reply:
I like "cascade" even better!